internal/characters/cyno/attack.go

```diff
@@ -0,0 +1,138 @@
+			ICDGroup:           combat.ICDGroupDefault,
+			Element:            attributes.Physical,
+			Durability:         25,
+			HitlagFactor:       0.01,
+			HitlagHaltFrames:   attackHitlagHaltFrame[c.NormalCounter][i],
+			CanBeDefenseHalted: attackDefHalt[c.NormalCounter][i],
+		}
+
+		c.Core.QueueAttack(
+			ai,
+			combat.NewCircleHit(c.Core.Combat.Player(), 0.5, false, combat.TargettableEnemy),
+			attackHitmarks[c.NormalCounter][i],
+			attackHitmarks[c.NormalCounter][i],
+		)
+	}
+
+	defer c.AdvanceNormalIndex()
+
+	return action.ActionInfo{
+		Frames:          frames.NewAttackFunc(c.Character, attackFrames),
+		AnimationLength: attackFrames[c.NormalCounter][action.InvalidAction],
+		CanQueueAfter:   attackHitmarks[c.NormalCounter][len(attackHitmarks[c.NormalCounter])-1],
+		State:           action.NormalAttackState,
+	}
+}
+
+const burstHitNum = 5
+
+var (
+	attackBFrames          [][]int
+	attackBHitmarks        = [][]int{{12}, {14}, {18}, {5, 14}, {40}}
+	attackBHitlagHaltFrame = [][]float64{{0.01}, {0.01}, {0.03}, {0.01, 0.03}, {0.05}}
+	attackBDefHalt         = [][]bool{{false}, {false}, {false}, {false, false}, {true}}
+)
+
+func init() {
+	// NA cancels (burst)
+	attackBFrames = make([][]int, burstHitNum)
+	attackBFrames[0] = frames.InitNormalCancelSlice(attackBHitmarks[0][0], 28) // N1 -> N2
+	attackBFrames[0][action.ActionAttack] = 16
+
+	attackBFrames[1] = frames.InitNormalCancelSlice(attackBHitmarks[1][0], 35) // N2 -> N3
+	attackBFrames[1][action.ActionAttack] = 31
+
+	attackBFrames[2] = frames.InitNormalCancelSlice(attackBHitmarks[2][0], 41) // N3 -> N4
+	attackBFrames[2][action.ActionCharge] = 39
+
+	attackBFrames[3] = frames.InitNormalCancelSlice(attackBHitmarks[3][0], 36) // N4 -> N5
+	attackBFrames[3][action.ActionAttack] = 27
+
+	attackBFrames[4] = frames.InitNormalCancelSlice(attackBHitmarks[4][0], 62) // N5 -> N1
+	attackBFrames[4][action.ActionCharge] = 500                                // illegal action
+}
+
+func (c *char) attackB(p map[string]int) action.ActionInfo {
+	c.tryBurstPPSlide(attackBHitmarks[c.NormalCounter][len(attackBHitmarks[c.NormalCounter])-1])
+
+	for i, mult := range attackB[c.NormalCounter] {
+		ai := combat.AttackInfo{
+			ActorIndex:         c.Index,
+			Abil:               fmt.Sprintf("Pactsworn Pathclearer %v", c.NormalCounter),
+			AttackTag:          combat.AttackTagNormal,
+			ICDTag:             combat.ICDTagNormalAttack,
+			ICDGroup:           combat.ICDGroupDefault,
+			Element:            attributes.Electro,
+			Durability:         25,
+			HitlagFactor:       0.01,
+			HitlagHaltFrames:   attackBHitlagHaltFrame[c.NormalCounter][i],
+			CanBeDefenseHalted: attackBDefHalt[c.NormalCounter][i],
+			Mult:               mult[c.TalentLvlBurst()],
+			FlatDmg:            c.Stat(attributes.EM) * 1.5, // this is A4
+		}
+		c.QueueCharTask(func() {
+			c.Core.QueueAttack(ai, combat.NewCircleHit(c.Core.Combat.Player(), 2, false, combat.TargettableEnemy), 0, 0)
```

Comment:
actual hitbox info is:
{{radius 2 circle}, {radius 2 circle}, {x=2.5 z=6.0 box}, {radius 2.5 circle, radius 2.5 circle}, {radius 3.5 circle}}